phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2077,6 +2077,13 @@ private PTable createTableInternal(CreateTableStatement statement, byte[][] spli
                     }
                 }
 
+                // Cannot set PHOENIX_TTL if parent has already defined it.
+                if (tableType == VIEW  && parent != null && parent.getPhoenixTTL() != PHOENIX_TTL_NOT_DEFINED) {
```

Comment:
We're doing this check and still doing a similar one on the server. Do we need both?

Reply:
This check is during the createTable flow. Moreover the current code does allow for easy client side checks for TableProperty changes during alter command.